internal/webdav/server.go

```diff
@@ -46,12 +46,21 @@ func NewServer(options ...Option) (*webdavServer, error) {
 	}, nil
 }
 
-func (s *webdavServer) Start(ctx context.Context, port string) {
+func (s *webdavServer) Start(ctx context.Context, port string, webdav_user string, webdav_pass string) {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		r = r.WithContext(context.WithValue(r.Context(), reqContentLengthKey, r.Header.Get("Content-Length")))
-		r = r.WithContext(context.WithValue(r.Context(), reqRangeKey, r.Header.Get("Range")))
-		s.handler.ServeHTTP(w, r)
+		username, password, _ := r.BasicAuth()
+
+		if username != webdav_user || password != webdav_pass {
+			w.Header().Set("WWW-Authenticate", `Basic realm="BASIC WebDAV REALM"`)
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte("401 Unauthorized"))
```

Comment:
```suggestion
			err:=w.Write([]byte("401 Unauthorized"))
			if err != nil {
				s.log.ErrorContext(r.Context(), "Error writting the response to the client", "err", err)
			}
```

This will fix the error :)